Source/Evolve/WarpXEvolve.cpp

```diff
@@ -286,9 +295,10 @@ WarpX::Evolve (int numsteps)
 
         m_particle_boundary_buffer->gatherParticles(*mypc, amrex::GetVecOfConstPtrs(m_distance_to_eb));
 
-        // Non-Maxwell solver: particles can move by an arbitrary number of cells
+        // Non-Maxwell solver or implicit: particles can move by an arbitrary number of cells
         if( electromagnetic_solver_id == ElectromagneticSolverAlgo::None ||
-            electromagnetic_solver_id == ElectromagneticSolverAlgo::HybridPIC )
+            electromagnetic_solver_id == ElectromagneticSolverAlgo::HybridPIC ||
+            evolve_scheme == EvolveScheme::ImplicitPicard)
```

Comment:
Ok. I was just thinking that the issue is whether particles can jump across processors. But yes, since the iterations assume locality the particles cannot go beyond the guard cells. I'll undo this change.